ColorzCore/Preprocessor/Directives/PoolDirective.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using ColorzCore.DataTypes;
+using ColorzCore.Lexer;
+using ColorzCore.Parser;
+using ColorzCore.Parser.AST;
+
+namespace ColorzCore.Preprocessor.Directives
+{
+    class PoolDirective : IDirective
+    {
+        public int MinParams => 0;
+        public int? MaxParams => 0;
+        public bool RequireInclusion => true;
+
+        public Maybe<ILineNode> Execute(EAParser p, Token self, IList<IParamNode> parameters, MergeableGenerator<Token> tokens)
+        {
+            foreach (List<Token> line in p.PooledLines)
```

Comment:
This will write the pooled lines in reverse order. Since order isn't meant to matter, it's probably fine, but I want to make sure it's intensional.

Reply:
Interesting quirk here, since the tokens are tagged with their original location, "relocating" them via pool shouldn't affect error parsing.

I think. Please test error messages to confirm they error on the AddToPool() and not on the #pool.

Reply:
Also are there errors with multiple pooled statements? There'd be a lack of newline tokens separating them, I'd think.

Reply:
Error messages are indeed being linked to the AddToPool invocation (and not the `#pool`). However it may be a good idea to add in a `note: pooled at xyz` or something but I don't really know how we would go about this?

Reply:
That would probably be extraneous/difficult to implement. (hm, if I were to implement it, I'd probably do some kind of algebraic effects on the error handler to wrap adding a "pooled at" message to the default handler... within the pool directive scope. but endsidethought)